subgraphs/uniswap-forks/src/common/creators.ts

```diff
@@ -349,37 +365,26 @@ class SwapTokens {
 export function getSwapTokens(
   token0: Token,
   token1: Token,
-  amount0: BigInt,
-  amount1: BigInt
+  amount0In: BigInt,
+  amount0Out: BigInt,
+  amount1In: BigInt,
+  amount1Out: BigInt
 ): SwapTokens {
   let tokenIn: Token;
   let tokenOut: Token;
   let amountIn: BigInt;
   let amountOut: BigInt;
 
-  if (isSameSign(amount0, amount1)) {
-    // if amounts are same sign, we can't determine the direction of the swap
-    log.critical(
-      "Swap amounts have same sign - cannot determine inbound and outbound tokens: token0: {} amount0: {}, token1: {} amount1: {}",
-      [
-        token0.id.toString(),
-        amount0.toString(),
-        token1.id.toString(),
-        amount1.toString(),
-      ]
-    );
-  }
-
-  if (amount0.gt(BIGINT_ZERO)) {
-    tokenIn = token0;
-    tokenOut = token1;
-    amountIn = amount0;
-    amountOut = amount1.times(BIGINT_NEG_ONE);
-  } else {
+  if (amount0Out.gt(BIGINT_ZERO)) {
     tokenIn = token1;
     tokenOut = token0;
-    amountIn = amount1;
-    amountOut = amount0.times(BIGINT_NEG_ONE);
+    amountIn = amount1In.minus(amount1Out);
+    amountOut = amount0In.minus(amount0Out).times(BIGINT_NEG_ONE);
```

Comment:
amount0 out is like the overflow number? Could make a comment on how you know it's neg